pkg/kubelet/hyper/hyperclient.go

```diff
@@ -533,147 +368,27 @@ func (c *HyperClient) CreatePod(podSpec *grpctypes.UserPod) (string, error) {
-	if err != nil {
-		return err
-	}
-
-	err = json.Unmarshal(body, &code)
-	if err != nil {
-		return err
-	}
-
-	if code != 0 {
-		return fmt.Errorf("Exit code %d", code)
-	}
-
-	return nil
-}
-
-func (c *HyperClient) GetTag() string {
-	dictionary := "0123456789abcdefghijklmnopqrstuvwxyz"
-
-	var bytes = make([]byte, 8)
-	rand.Read(bytes)
-	for k, v := range bytes {
-		bytes[k] = dictionary[v%byte(len(dictionary))]
-	}
-	return string(bytes)
-}
-
-func (c *HyperClient) hijack(method, path string, hijackOptions hijackOptions) error {
-	var params io.Reader
-	if hijackOptions.data != nil {
-		buf, err := json.Marshal(hijackOptions.data)
-		if err != nil {
-			return err
-		}
-		params = bytes.NewBuffer(buf)
-	}
 
-	if hijackOptions.tty {
-		in, isTerm := term.GetFdInfo(hijackOptions.in)
-		if isTerm {
-			state, err := term.SetRawTerminal(in)
-			if err != nil {
-				return err
-			}
+	ctx, cancel := getContextWithTimeout(hyperContextTimeout)
+	defer cancel()
 
-			defer term.RestoreTerminal(in, state)
-		}
-	}
-	req, err := http.NewRequest(method, fmt.Sprintf("/v%s%s", hyperMinimumVersion, path), params)
+	response, err := c.client.PodCreate(ctx, &request)
 	if err != nil {
-		return err
-	}
-
-	req.Header.Set("User-Agent", "kubelet")
-	req.Header.Set("Content-Type", "text/plain")
-	req.Header.Set("Connection", "Upgrade")
-	req.Header.Set("Upgrade", "tcp")
-	req.Host = HYPER_ADDR
-
-	dial, err := net.Dial(HYPER_PROTO, HYPER_ADDR)
-	if err != nil {
-		return err
-	}
-
-	clientconn := httputil.NewClientConn(dial, nil)
-	defer clientconn.Close()
-
-	clientconn.Do(req)
-
-	rwc, br := clientconn.Hijack()
-	defer rwc.Close()
-
-	errChanOut := make(chan error, 1)
-	errChanIn := make(chan error, 1)
-	exit := make(chan bool)
-
-	if hijackOptions.stdout == nil && hijackOptions.stderr == nil {
-		close(errChanOut)
-	}
-	if hijackOptions.stdout == nil {
-		hijackOptions.stdout = ioutil.Discard
-	}
-	if hijackOptions.stderr == nil {
-		hijackOptions.stderr = ioutil.Discard
-	}
-
-	go func() {
-		defer func() {
-			close(errChanOut)
-			close(exit)
-		}()
-
-		_, err := io.Copy(hijackOptions.stdout, br)
-		errChanOut <- err
-	}()
-
-	go func() {
-		defer close(errChanIn)
-
-		if hijackOptions.in != nil {
-			_, err := io.Copy(rwc, hijackOptions.in)
-			errChanIn <- err
-
-			rwc.(interface {
-				CloseWrite() error
-			}).CloseWrite()
-		}
-	}()
-
-	<-exit
-	select {
-	case err = <-errChanIn:
-		return err
-	case err = <-errChanOut:
-		return err
+		return "", err
 	}
 
-	return nil
+	return response.PodID, nil
 }
 
-func (client *HyperClient) Attach(opts AttachToContainerOptions) error {
+func (c *HyperClient) Attach(opts AttachToContainerOptions) error {
 	if opts.Container == "" {
 		return fmt.Errorf("No Such Container %s", opts.Container)
 	}
 
-	stream, err := client.client.Attach(context.Background())
+	ctx, cancel := getContextWithTimeout(hyperContextTimeout)
+	defer cancel()
+
+	stream, err := c.client.Attach(ctx)
```

Comment:
Should use `context.Background()` instead because we can't predict how long will users interact with containers.